README.md

```diff
@@ -709,6 +722,149 @@ options on to the custom `TaskQueue` implementation. (Note that because the
 queue options are set as a property on the task, tasks with queue
 options cannot be submitted as JavaScript primitives).
 
+## Custom Worker Scheduling
+
+Piscina provides a default scheduling algorithm that attempts to spawn threads
+based on the max number of threads, and the possibility of a task being cancelled.
+
+This algorithm can be overriden by passing a Custom Scheduler to the `Piscina`
+constructor.
+
+Custom schedulers must implement the `Scheduler` interface, described below:
+```ts
+class TaskScheduler {
+  size: number = 0
+
+  // The constructor will receive the max number of concurrent task per worker
+  // as passed when instantiating Piscina
+  constructor (concurrentTasksPerWorker: number) {
+  }
+
+  [Symbol.iterator] (): IterableIterator<PiscinaWorker> {
+    // Meant to enable Piscina iterate over the workers for different purposes
```

Comment:
What's this for? It kind of breaks en encapsulation of workers.

Reply:
It's to iterate over the threads spawned. I can actually thin this and replace it with an explicit API, I'll do that instead; anyways this is used only in a single place 👍